source/common/router/router.cc

```diff
@@ -1804,6 +1808,10 @@ void Filter::doRetry(bool can_send_early_data, bool can_use_http3) {
     downstream_headers_->setEnvoyAttemptCount(attempt_count_);
   }
 
+  if (include_timeout_retry_header_in_request_) {
+    downstream_headers_->setEnvoyIsTimeoutRetry(is_timeout_retry? "true" : "false");
```

Comment:
maybe it would be better to avoid setting the header altogether when it was not a timeout? as opposed to setting it with "false"?

Reply:
also put a space before the `?`

Reply:
when I posted the initial comment I thought this was going on _all_ requests, not just retries. It seems reasonable to me to include the header for all retries